extension.js

```diff
@@ -415,15 +417,33 @@ const ClipboardIndicator = Lang.Class({
         }
     },
 
+    _cancelNotification: function() {
+        if (this.clipItemsRadioGroup.length >= 2) {
+            let clipSecond = this.clipItemsRadioGroup.length - 2;
+            let previousClip = this.clipItemsRadioGroup[clipSecond];
+            Clipboard.set_text(CLIPBOARD_TYPE, previousClip.clipContents);
+            previousClip.setOrnament(PopupMenu.Ornament.DOT);
+            previousClip.icoBtn.visible = false;
+            previousClip.currentlySelected = true;
+        } else {
+            Clipboard.set_text(CLIPBOARD_TYPE, "");
+        }
+        let clipFirst = this.clipItemsRadioGroup.length - 1;
+        this._removeEntry(this.clipItemsRadioGroup[clipFirst]);
+    },
+
     _showNotification: function (message) {
         let notification = null;
 
         this._initNotifSource();
 
         if (this._notifSource.count === 0) {
             notification = new MessageTray.Notification(this._notifSource, message);
+            if (message == _("Copied to clipboard")) {
+                notification.addAction( _('Cancel') , Lang.bind(this, function() {this._cancelNotification();}) );
+            }
         }
-        else {
+        else if (message != _("Copied to clipboard")) {
```

Comment:
In case of a (second) copy to clipboard while there is a notification, the action button appears twice (even three times for two copies while there is a notification).
To prevent it, I tried to use the function "clearActions()" that I found in specification but it seems not to work, at least with this orthograph or by this code `notification.clearActions();`.
The best would be to succeed to use this method.

I added this condition waiting a better solution but you are right that's not good enough.

I hope someone reach a good solution.